ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Storages.resource

```diff
@@ -66,53 +67,65 @@ Storage Size Should Be
-        ${displayed_size}=    Get Text    ${bound_storage_size_el}
-        Run Keyword And Continue On Failure     Should Be Equal As Strings    ${displayed_size}    ${size}Gi
+        ...    ${STORAGE_SECTION_XP}//td[contains(@data-label,"Size")]//div[strong[text()="Size"]]/div/div[3]/div    # robocop: disable
+        Wait Until Page Contains Element    ${bound_storage_size_el}    timeout=2m
+        ${displayed_sizes}=    Get All Text Under Element    ${bound_storage_size_el}
+        ${matched_sizes}=    Get All Strings That Contain    ${displayed_sizes}    ${size}Gi
+        Run Keyword And Continue On Failure    Should Not Be Empty    ${matched_sizes}
         ${rc}  ${oc_object_size}=    Run And Return Rc And Output
         ...    oc get pvc -n ${namespace} -o jsonpath='{.items[?(@.metadata.annotations.openshift\\.io/display-name=="${name}")].status.capacity.storage}'    # robocop: disable
         Should Be Equal As Integers    ${rc}    0
         ...    An error occurred during the check of PVC ${name} '.status.capacity.storage' value!
-        Run Keyword And Continue On Failure     Should Be Equal As Strings    ${oc_object_size}    ${size}Gi
+        Run Keyword And Continue On Failure     Should Contain    ${oc_object_size}    ${size}Gi
     ELSE
         ${error_msg}=    Catenate    The PVC is in a '${pvc_status_phase}' state which we don't expect for this
         ...    scenario. We expect either 'Pending' or 'Bound' states here. Please, either complete the
         ...    implementation or investigate the test failure.
         Fail    ${error_msg}
         RETURN    1
     END
-    # This check is common for both "Pending" and "Bound" states of PVC.
+    # This check is common for both "Pending" and "Bound" states of PVC, and it verifies storage size directly via OC CLI
     ${rc}  ${oc_object_size}=    Run And Return Rc And Output
-    ...    oc get pvc -n ${namespace} -o jsonpath='{.items[?(@.metadata.annotations.openshift\\.io/display-name=="${name}")].spec.resources.requests.storage}'    # robocop: disable
+    ...    oc get pvc -n ${namespace} -o jsonpath="{.items[?(@.metadata.annotations['openshift\\.io/display-name']=='${name}')].spec.resources.requests.storage}"    # robocop: disable
     Should Be Equal As Integers    ${rc}    0
     ...    An error occurred during the check of PVC ${name} '.spec.resources.requests.storage' value!
     Run Keyword And Continue On Failure     Should Be Equal As Strings    ${oc_object_size}    ${size}Gi
 
 Create PersistentVolume Storage
     [Documentation]    Create a PersistenVolume storage in DS Project details page
-    [Arguments]    ${project_title}    ${name}    ${description}    ${size}    ${connected_workbench}=${NONE}    ${press_cancel}=${FALSE}
-    Click Button    ${STORAGE_ADD_BTN_1_XP}
+    [Arguments]    ${project_title}    ${name}    ${description}    ${size}    ${connected_workbench}=${NONE}
+    ...    ${press_cancel}=${FALSE}    ${existing_storage}=${FALSE}
```

Comment:
what's the benefit of re-using a pvc when calling the kw to create one?

Reply:
imo it makes the tests less predictable and may hide other issues (maybe test env issues). Unless we explicitly want the PVC to be already present for so efficiency reasons of the tests maybe. In this particular case, why the PVC would be already present before?

Reply:
It's an improvement of this Keyword, like we have with other keywords. 
 This opens an option for more test logic where we can edit existing PVC, if it exists (and create if it is not).
Search in code for `${existing_` and you'll other keywords that uses that logic.